staging/src/k8s.io/apimachinery/pkg/api/errors/errors.go

```diff
@@ -594,11 +715,12 @@ func IsTooManyRequests(err error) bool {
 // the request entity is too large.
 // It supports wrapped errors.
 func IsRequestEntityTooLargeError(err error) bool {
-	if ReasonForError(err) == metav1.StatusReasonRequestEntityTooLarge {
+	reason, code := reasonAndCodeForError(err)
+	if reason == metav1.StatusReasonRequestEntityTooLarge {
 		return true
 	}
-	if status := APIStatus(nil); errors.As(err, &status) {
-		return status.Status().Code == http.StatusRequestEntityTooLarge
+	if _, ok := knownReasons[reason]; !ok && code == http.StatusRequestEntityTooLarge {
+		return true
 	}
```

Comment:
since this was already checking code without regard for reason, continue doing so and add a comment noting why this is different than the other helpers